clar2wasm/tests/wasm-generation/contracts.rs

```diff
@@ -538,4 +538,87 @@ proptest! {
             }))),
         );
     }
+
+    #[test]
+    fn contract_dynamic_call_use_all_args(
+        (tys, values)
+            in prop::collection::vec(
+                prop_signature().prop_ind_flat_map2(|ty| PropValue::from_type(ty.clone())),
+                1..=20
+            )
+            .prop_map(|arg_ty| arg_ty.into_iter().unzip::<_, _, Vec<_>, Vec<_>>())
+            .no_shrink(),
+        err_type in prop_signature(),
+    ) {
+        // first contract
+        let first_contract_name = "foo".into();
+        let mut function_types = String::new();
+        let mut function_arguments = String::new();
+        for (name, ty) in ('a'..).zip(tys.iter()) {
+            let ty = type_string(ty);
+            write!(function_arguments, "({name} {ty}) ").unwrap();
+            function_types += &(ty + " ");
+        }
+        let expected_res: PropValue = Value::from(
+            TupleData::from_data(
+                ('a'..)
+                    .map(|c| ClarityName::try_from(c.to_string()).unwrap())
+                    .zip(values.iter().cloned().map(Value::from))
+                    .collect(),
+            )
+            .unwrap(),
+        )
+        .into();
+
+        let first_snippet = format!(
+            r#"
+                    (define-trait foo-trait (
+                        (foofun ({function_types}) (response {} {}))
+                    ))
+
+                    (define-public (foofun {function_arguments})
+                        (ok {expected_res})
```

Comment:
Isn't this returning a hardcoded result? vs using the function args

Reply:
oof yes, no idea what I did there.

Reply:
Fixed in [35c3dd7](https://github.com/stacks-network/clarity-wasm/pull/484/commits/35c3dd770aacfffb710dc8e74dbe5aaf8c0113ac)